### ISSUE TITLE
$ObjMap extend type of every value

### ISSUE THREAD

**Issue by macrozone** [facebook/flow#6343]
I have this type `Test`

```
type Test = {
   foo: string
}
```

and now i want to create a type that accepts both of this objects: 

```
const a = {
   foo: "my string"
}

//and

const b = {
   foo: ["my string in an array"]
}

// but it should reject

const c = {
   foo: [3]
}
```

I tried `$ObjMap` like this:

```
type Test = {
   foo: string
}

type TypeOrArray = <V>(V) => Array<V> | V
   
type ValuesAsArray<T> = $ObjMap<T, TypeOrArray>

type MyType = ValuesAsArray<Test>
````

 but with strange results:

https://flow.org/try/#0C4TwDgpgBAKhDOwoF4oG8BQVsDMD2eAXFIgE4CWAdgOYYC+Gjoks4EA8qQIKmkCGIFFAA8ANQB8AClEBKFOKg9+IMQoA+UUVmwZm0UXwA2AVwRd4SgcJgLUAEnYAjAFYBZPmGsAaVpE6WQcW0dPShXEBg2IQMTMwteKzhEIIwAelSSAAs8Y0MAEyg+AGMiiDAkYEzyeAwivEpEQsJwyJZUTGwofCIoACIAW0EyKmpe+kZ0wsoC+Gz8wpKyisy+YFr6xsdmiKj24O7iAG0BoeAKGigqKcKEkF6AXXHJ2Zz50ghnCCLl1cL4LoIl3+lDwSD4JDOI3WDSQRW2rWge06BygAGZxp0oGkMi9cgUjPA8FB3p9vlBKr8+P9ukCoCCwRDztQALR8W7Qxp5eG7dD7AhHVGPBg6bQYIA

Is there something i do wrong here?

**Comment by mrkev** [comment]
What is wrong? It looks to me like it's working fine

**Comment by macrozone** [comment]
@mrkev sorry, i got confused by the error reporting. The issue seems to be fixed in flow 0.72 (but existed in previous versions)
